src/reporting.ts

```diff
@@ -56,18 +54,20 @@ export const getNotesInMarkdown = (): string => {
 };
 
 export const generateMarkdownReport = (): void => {
-    const workspaceFolder = vscode.workspace.rootPath;
-    // TODO: What if there's no workspace?
-    if (workspaceFolder) {
-        // TODO: Remove this hardcoded string, it should be a configuration
-        const extensionDirPath = path.join(workspaceFolder, '.vscode', 'code-annotation');
-        const extensionFilePath = path.join(extensionDirPath, 'summary.md');
-        let content = getNotesInMarkdown();
-        fs.writeFileSync(extensionFilePath, content);
-        var openPath = vscode.Uri.file(extensionFilePath);
-        vscode.workspace.openTextDocument(openPath).then(doc => {
-            vscode.window.showTextDocument(doc).then(editor => {
-            });
+    const newFile = vscode.Uri.parse('untitled:summary.md');
+
+    vscode.workspace.openTextDocument(newFile).then(summaryFile => {
+        const edit = new vscode.WorkspaceEdit();
+        let notesSummary = getNotesInMarkdown();
+
+        edit.insert(newFile, new vscode.Position(0, 0), notesSummary);
```

Comment:
```suggestion
        const existingContentRange = new vscode.Range(new vscode.Position(0, 0),
                                     new vscode.Position(summaryFile.lineCount + 1, 0));
        edit.replace(newFile, existingContentRange, notesSummary);
```

Turns out we can get the previous range from the summaryFile returned. In such case I'm creating a range from the beginning `0, 0`, until the last line + 1 (just to enforce getting the whole content).

Thank you a lot for the work on this @taismassaro !

Reply:
cool! that’s what I understood I should do, but didn’t find a way to do it 😬 thanks for sharing! 😊